### PR TITLE
[stable/traefik] Upgrade traefik version to 1.3.4, fix toleration with 1.5 cluster and give traefik access to secrets for the k8s basic authentication

### PR DESCRIPTION
**Pull request by djsly**
if you guys could pick this up, it would be appreciated! :) 

@krancour @viglesiasce 

Thanks!

### CODE REVIEW

**Comment by kachkaev** [comment]
@djsly can you also add this to your PR? https://github.com/kubernetes/charts/pull/1452 I'll close mine then.

**Comment by djsly** [comment]
@kachkaev : done! :) 
I have added your commit from #1452

**Comment by krancour** [comment]
In the interest of tractability, it might be better for these three changes to each be their own PR. I know one of the features was (#1452) and seems to have been added into here because @kabakaev is eager.

If you need more convincing, two out of three of these changes I could LGTM right away, but one requires a little more research on my part to confirm it. By bundling the three together, the two that could be merged sooner get held up.

**Comment by djsly** [comment]
Hello @krancour , one of the main reason I kept all three changes in three different commits was to help identifying the three changes seperatly. With the dependency of having to update the `Charts.yaml --> version: ` with every PR, it makes it that we often end up having to rebase depending on the order the PR are accepted. 

I will try to create multiple PR in the future.

**Comment by krancour** [comment]
>  one of the main reason I kept all three changes in three different commits was to help identifying the three changes seperatly. With the dependency of having to update the Charts.yaml --> version: with every PR, it makes it that we often end up having to rebase depending on the order the PR are accepted.

Yep. Understood. Let me say, I appreciate that the three changes were three separate commits. That does make review easier. And the need to rebase depending on the merge order (to get version numbering right) is most definitely a pain. No argument there. But alas... the three commits are still bundled together in one PR which means the fates of all three changes are tied together until all three can be resolved (LGTM'ed or vetoed).

At any rate... I don't want to waste more of everyone's time than I already have bikeshedding about PRs. I'm just going to buckle down and look closer at the one change I wasn't sure about so that I can try to LGTM this whole thing.

File: stable/traefik/Chart.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v1
 name: traefik
-version: 1.7.0
-appVersion: 1.3.1
+version: 1.8.0
```
Comment (krancour) [inline stable/traefik/Chart.yaml]: Presumably, the delta between Traefik `1.3.1` and `1.3.3` is only bug fixes. If that's the case, the net effect on the chart is also only bug fixes. So shouldn't the new chart version be `1.7.1`?

Comment (djsly) [reply stable/traefik/Chart.yaml]: sure thing. I wasn't fully aware of the logic behind the chart `Version`. I agree that it should follow the version of the image. e.g. Major upgrade of `appVersion`  will mean a major upgrade of `version`

Comment (djsly) [reply stable/traefik/Chart.yaml]: Fixed

**Comment by krancour** [comment]
This LGTM, but I'm not sure why the commits jumped from 3 to 6 after your recent fix. If you've got a minute, can you just clean this up? Maybe start a new local branch from master, cherry pick the three relevant changes and then force push from that local branch to this one? It will be cleaner, easier to follow history.

**Comment by djsly** [comment]
ohhh sorry i must of rebased from the wrong branch...

**Comment by djsly** [comment]
Fixed @krancour , back to 3 commits.

**Comment by krancour** [comment]
Totally LGTM now. cc @viglesiasce @lachie83

**Comment by kachkaev** [comment]
ping @viglesiasce @lachie83 :–)

**Comment by kachkaev** [comment]
Ouch. The PR is still not merged after 2 weeks. Ping @prydonius @unguiculus :pray:

**Comment by kachkaev** [comment]
@djsly looks like it's time to upgrade traefik to `1.3.4` :smile:

**Comment by djsly** [comment]
@kachkaev done.
@krancour what is the process of getting those PR merged ? this is my firsts involvement with the public charts, I getting a feeling that we might be better sticking to local/<charts> when using helm even for public charts.

**Review by prydonius — CHANGES_REQUESTED** [review]
Hi @djsly, apologies for the slow review. Not sure if I'm missing something but it doesn't look like the imageTag in values.yaml has been updated to 1.3.4, which means that this will stay on 1.3.1. https://github.com/djsly/charts/blob/ca9627cdcb28f72fc3ae22cf73820ab215d2f3ca/stable/traefik/values.yaml#L3

**Comment by djsly** [comment]
@prydonius thanks... Sorry about that, I got confused by the `appVersion: 1.3.4` field in the Charts.yaml. good catch!

**Comment by prydonius** [comment]
@djsly no problem, the appVersion is just for metadata (it doesn't necessarily relate to the image tags used, as there could be multiple images in a given application).

**Comment by kachkaev** [comment]
It's probably because 1.3.4 is not yet in https://hub.docker.com/_/traefik/. It's in https://hub.docker.com/r/containous/traefik/ though. Explanation of the difference: https://github.com/containous/traefik-library-image/issues/3#issuecomment-318429169

The official upgrade should not take long, because the PR has already been merged: https://github.com/docker-library/official-images/pull/3243

**Comment by prydonius** [comment]
@kachkaev thanks for the explanation. @djsly up to you whether you want to put this back to 1.3.3 to get this in now and update to 1.3.4 later in a separate PR, or just wait on this until 1.3.4 is available.

**Comment by djsly** [comment]
I dont' mind waiting 1 extra day assuming @prydonius will be ready to issue a /retest once ready. me and @kachkaev can monitor for the release of 1.3.4

**Comment by prydonius** [comment]
@djsly sgtm - I believe you also have the ability to issue a `/retest` or `/test all` :)

**Comment by djsly** [comment]
sweet!

**Comment by kachkaev** [comment]
1.3.4 is there: https://hub.docker.com/r/library/traefik/tags/  🎉

**Comment by kachkaev** [comment]
@prydonius your move I believe :–)

**Review by prydonius — APPROVED** [review]

**Comment by prydonius** [comment]
Thanks @djsly and @kachkaev!!

**Comment by kachkaev** [comment]
I'll submit a tiny PR now to replace `secret` with `secrets`

**Pull request merged**